### PR TITLE
[massive] Internal refactor to use new react context api

### PR DESCRIPTION
**Pull request by 3rd-Eden**
As per title,

Internal refactor to use the new context api for validation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,82 +1,89 @@
...
 
-
-# Install
+## Install
 
 This package is available on `npm`. Install it using:
 
 ```sh
-npm install react-validation-context --save
+npm install --save react-validation-context
```
Comment (msluther) [inline README.md:15]: Could drop `--save` altogether since it's the default now.

File: package.json [hunk 2]
```diff
@@ -49,35 +41,38 @@
...
+    "nyc": "^14.1.1",
     "prop-types": "^15.7.2",
     "react": "^16.3.0",
+    "react-dom": "^16.8.6",
     "react-test-renderer": "^16.3.0",
+    "setup-env": "^1.2.2",
     "sinon": "^7.2.5"
   },
   "peerDependencies": {
     "react": "^16.4.0 || ^17.0.0"
```
Comment (msluther) [inline package.json:73]: that's pretty ambitious of it considering there's no way to test it or be sure it's going to work 😂

File: test/index.test.js [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+          { child && child }
+        </Validate>
+      ), {
+        lifecycleExperimental: true
+      });
+
+      //
+      // We need "unpeel" the HOC layers to actually access the right component.
+      //
+      wrapper = context.dive().dive().dive();
```
Comment (msluther) [inline test/index.test.js:142]: bwoop bwoop bwoop, Imma submarine! 😂

File: validate.js [hunk 4]
```diff
@@ -44,40 +52,34 @@ export default class Validate extends Validates {
...
-        if (isValid === undef) {
-          delete valids[name];
-        } else {
-          valids[name] = isValid;
-        }
-        this.setState({ valids, validates: validate(valids) });
-      }
-    };
+  processValidChange(name, isValid) {
+    const { valids } = this.state;
```
Comment (msluther) [inline validate.js:66]: does that need to be shallow cloned? you're technically modifying the object that's already in state and putting the same object reference back in, which might be okay. i guess you only really need to process the state change if `validates` changes?

Comment (msluther) [reply validate.js:66]: Maybe worth a comment so it doesn't look like a bug?

**Review by msluther — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (3rd-Eden) [reply README.md:15]: DID YOU JUST ASSUME MY NPM VERSION??

**Review by burgessa23 — APPROVED** [review]

File: validate.js (hunk 4 above)
Comment (3rd-Eden) [reply validate.js:66]: Good find, this was still the original code, so it works, but i'll make clone of the object.

Comment (3rd-Eden) [reply validate.js:66]: Good find, this was still the original code, so it works, but i'll make clone of the object.

Comment (3rd-Eden) [reply validate.js:66]: Already wasted too much time on this, we can fix this in another sprint :joy:

**Pull request merged**